### PR TITLE
Beckdave logger

### PR DESCRIPTION
**Pull request by beckdave**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,6 +72,7 @@ include_directories(inc)
 file(GLOB SOURCES "src/*.cpp" "inc/*.h"
                   "src/helpers/*.cpp" "inc/helpers/*.h"
                   "src/json/*.cpp" "inc/json/*.h"
+                  "src/logger/*.cpp" "inc/logger/*.h"
```
Comment (mnaveedb) [inline CMakeLists.txt:75]: Can we change it. Not using wildcard?
But using full file name.

File: inc/helpers/BexDefines.h [hunk 2]
```diff
@@ -30,16 +30,25 @@ namespace finalmq {
...
 static const int INVALID_FD = -1;
 
+}   // namespace finalmq
 
-#if defined(WIN32) || defined(__MINGW32__)
 
+#if defined(WIN32) || defined(__MINGW32__)
 #else
-
 typedef int					SOCKET;
```
Comment (mnaveedb) [inline inc/helpers/BexDefines.h:38]: Well we can using more modern syntax here. 
ussing SOCKET = int;

File: inc/helpers/BexDefines.h [hunk 3]
```diff
@@ -30,16 +30,25 @@ namespace finalmq {
...
+
+#if defined(WIN32)
+#define METHODNAME  __FUNCTION__
+#else
+#define METHODNAME  __FUNCTION__
 #endif
 
-}   // namespace finalmq
+
+#define TOKENPASTE(x, y) x ## y
```
Comment (mnaveedb) [inline inc/helpers/BexDefines.h:49]: consider using constexpr function here

File: inc/logger/LogStream.h [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#pragma once
+
+#include "Logger.h"
+#include "helpers/BexDefines.h"
+#include <string>
+#include <sstream>
+
+
+
+#define STREAM_BASIC(level)     static const finalmq::LogContext TOKENPASTE2(context, __LINE__){level, MODULENAME, METHODNAME, __FILE__, __LINE__}; finalmq::LogStream::getStream(TOKENPASTE2(context, __LINE__))
```
Comment (mnaveedb) [inline inc/logger/LogStream.h:32]: Would it produce a global state?

File: inc/helpers/BexDefines.h (hunk 2 above)
Comment (beckdave) [reply inc/helpers/BexDefines.h:38]: I changed it

File: inc/logger/LogStream.h (hunk 4 above)
Comment (beckdave) [reply inc/logger/LogStream.h:32]: static const LogContext context__LINE__
is a local variable where ever a log entry is defined. To have unique variable names the line number is part of the variable name.

**Pull request merged**